__device-tests__/gutenberg-editor-writing-flow.test.js

```diff
@@ -0,0 +1,104 @@
+/**
+ * Internal dependencies
+ */
+const { blockNames } = editorPage;
+
+describe( 'Gutenberg Editor - Writing Flow', () => {
+	describe( 'TC007 - Test format detection under the cursor', () => {
+		it( 'checks that the proper format buttons are selected when the cursor is under', async () => {
+			const paragraphHTML = `<!-- wp:paragraph -->
+            <p><span accessibility-id="bold"><strong>bold</strong></span>
+            <span accessibility-id="bold-italic"><strong><i>bold-italic</i></strong>.</span>
+            <span accessibility-id-"strikethrough"><s>strikethrough</s></span></p>
+            <!-- /wp:paragraph -->`;
+
+			// On a rich-text based component, add bold, italic, strikethrough and link formatted text
+			await editorPage.addNewBlock( blockNames.paragraph );
+			const paragraphBlockElement = await editorPage.getTextBlockAtPosition(
+				blockNames.paragraph
+			);
+
+			await editorPage.typeTextToTextBlock(
+				paragraphBlockElement,
+				paragraphHTML
+			);
+
+			// Select bold text
+			// TODO: verify if this is the best method to select text formatting
+			await editorPage.elementsByAccessibilityId( 'bold' );
+
+			// Check that the proper format buttons get selected
+			// TODO: Identify best method to identify format buttons
+
+			// TODO: Invoke assertion or snapshot here?
+
+			// TODO: Repeat for bold-italic and strikethrough
+
+			await editorPage.removeBlock();
+		} );
+	} );
+
+	describe( `TC009 - Test autocorrection doesn't apply formatting to Heading`, () => {
+		it( 'checks that formatting is not applied to autocorrected text', async () => {
+			const headingText = `<p>Mispelled</p>`;
+
+			// Add a Heading block without any formatting applied.
+			await editorPage.addNewBlock( blockNames.heading );
+			const headingBlockElement = await editorPage.getTextBlockAtPosition(
+				blockNames.heading
+			);
+
+			// Type a sentence with a word misspelled that will be autocorrected by the editor
+			await editorPage.typeTextToTextBlock(
+				headingBlockElement,
+				headingText
+			);
+
+			// Highlight the word that was autocorrected
+			// Simulated double tap. TODO: Verify if this is the best method to highlight text
+			await editorPage.tapElement( headingBlockElement );
+			await editorPage.tapElement( headingBlockElement );
+
+			// Check that the word has no formatting applied
+			const selectedText = await editorPage.getTextForParagraphBlockAtPosition(
+				1
+			);
+			// TODO: verify if a snapshot should be used in place of an assertion, or both
+			expect( selectedText ).toMatch( headingText );
```

Comment:
We _could_ use a snapshot. However, I do not think the current assertion is incorrect. It is a simple assertion and easy to maintain. 

IMO, snapshots are incredibly helpful for asserting against expectations that are non-trivial to write (writing lots of markup) or change frequently (markup could change). 

If we did use a snapshot here, I might select an inline snapshot given the output is so small.